src/coreclr/hosts/corerun/corerun.hpp

```diff
@@ -572,7 +572,7 @@ namespace pal
             // Make sure if we have an assembly with multiple extensions present,
             // we insert only one version of it.
             if (should_add(entry->d_name))
-                file_list << directory << dir_delim << entry->d_name << env_path_delim;
+                file_list << directory << entry->d_name << env_path_delim;
```

Comment:
What about DT_UNKNOWN case in line 557?

Reply:
That doesn't matter, because DT_UNKNOWN and DT_LNK are only used to check whether file exist and path with double `//` is correct in Unix, plus it's not affecting anything outside `corerun`. 

```C++
case DT_LNK:
case DT_UNKNOWN:
    {
        string_t full_filename;
        full_filename.append(directory);
        full_filename.append(1, pal::dir_delim);
        full_filename.append(entry->d_name);
        if (!does_file_exist(full_filename.c_str()))
            continue;
    }
    break;
```

But yes, `full_filename.append(1, pal::dir_delim);` can also be removed, because it's not needed.

Reply:
I've decided to remove it.